First api wrapper function

pyMike opened the pull request:
My first module submitted for review in Python for the open contracting community.
Purpose to allow users to easily access the Restful API that serves URL links to data.

pyMike committed:
Just testing

pyMike committed:
first wrapper api

pyMike commented:
This module contains examples of use cases.
Note, this module is a work in progress and gets the job done but 
it's by no means polished code.
Feel free to make suggestions for improvement.
Regards

birdsarah commented:
Hi Michael, great this all looks like its coming through as expected.

Stay on your feature branch and:
- delete the file experiments/test.txt (git rm experiments/test.txt)
- update the OCDS_API.py file to have one of thet functions we agreed. I believe we said save_csvs_by_dataset

Once you update your feature branch and then push to github, he pull request will automatically update with these changes.

pyMike commented:
Thanks for the advice Sarah

It gives me confidence I'm doing the right thing.

All the Best

Mike

On Fri, Jul 4, 2014 at 9:36 AM, Sarah Bird notifications@github.com wrote:

> Hi Michael, great this all looks like its coming through as expected.
> 
> Stay on your feature branch and:
> - delete the file experiments/test.txt (git rm experiments/test.txt)
> - update the OCDS_API.py file to have one of thet functions we agreed.
>   I believe we said save_csvs_by_dataset
> 
> Once you update your feature branch and then push to github, he pull
> request will automatically update with these changes.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/open-contracting/ocdata-blend/pull/4#issuecomment-48044405
> .

pyMike committed:
Added get_links for filtering URLs within json_dict data structure. Also deleted test.txt

pyMike committed:
Added getting more metadata from each link/file and extending/augmenting the json_dict data structure to include the augmented info like filename, size, url and http instance information from the server.

pyMike commented:
Not sure the last two pull requests were sent since
I don't remember seeing a green button for the pull request.
Please let me know you've got them.
They should be listed as...
1) Added get_links...
2) Added getting more metadata...

Thanks
Mike

pyMike commented:
Hi Sarah

Not sure the last two pull requests were sent since
I don't remember seeing a green button for the pull request.
Please let me know you've got them.
They should be listed as...
1) Added get_links...
2) Added getting more metadata...

Thanks
Mike

On Fri, Jul 4, 2014 at 9:50 AM, Michael Glaude michael.glaude@gmail.com
wrote:

> Thanks for the advice Sarah
> 
> It gives me confidence I'm doing the right thing.
> 
> All the Best
> 
> Mike
> 
> On Fri, Jul 4, 2014 at 9:36 AM, Sarah Bird notifications@github.com
> wrote:
> 
> > Hi Michael, great this all looks like its coming through as expected.
> > 
> > Stay on your feature branch and:
> > - delete the file experiments/test.txt (git rm experiments/test.txt)
> > - update the OCDS_API.py file to have one of thet functions we
> >   agreed. I believe we said save_csvs_by_dataset
> > 
> > Once you update your feature branch and then push to github, he pull
> > request will automatically update with these changes.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/open-contracting/ocdata-blend/pull/4#issuecomment-48044405
> > .

pyMike commented:
Hi Sarah

Forgot to mention, that the next step will be the saving/downloading of the
files as requested.
The last pull request goes up to the point where the file information is
available for saving.
I had to do that first in order to get the filename for saving.

I'm taking a short brake and will get on the saving next.

Mike

On Fri, Jul 4, 2014 at 2:15 PM, Michael Glaude michael.glaude@gmail.com
wrote:

> Hi Sarah
> 
> Not sure the last two pull requests were sent since
> I don't remember seeing a green button for the pull request.
> Please let me know you've got them.
> They should be listed as...
> 1) Added get_links...
> 2) Added getting more metadata...
> 
> Thanks
> Mike
> 
> On Fri, Jul 4, 2014 at 9:50 AM, Michael Glaude michael.glaude@gmail.com
> wrote:
> 
> > Thanks for the advice Sarah
> > 
> > It gives me confidence I'm doing the right thing.
> > 
> > All the Best
> > 
> > Mike
> > 
> > On Fri, Jul 4, 2014 at 9:36 AM, Sarah Bird notifications@github.com
> > wrote:
> > 
> > > Hi Michael, great this all looks like its coming through as expected.
> > > 
> > > Stay on your feature branch and:
> > > - delete the file experiments/test.txt (git rm experiments/test.txt)
> > > - update the OCDS_API.py file to have one of thet functions we
> > >   agreed. I believe we said save_csvs_by_dataset
> > > 
> > > Once you update your feature branch and then push to github, he pull
> > > request will automatically update with these changes.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/open-contracting/ocdata-blend/pull/4#issuecomment-48044405
> > > .

pyMike committed:
Automated Downloading and Saving files to local file system in current directory.

pyMike commented:
Hi Sarah
The saving function you requested is implemented.
Tomorrow, I'm planning on implementing the "save_csvs_size_limit" function
and improving the filename extraction which is currently rudimentary.
Just wanted to get version 1 of everything out there in case anyone needs it.
Hope all is going well for you.
Mike

birdsarah commented:
Hi Michael,

In order to review this, this should just be the code that I am reviewing, either with a docstring to document a method or a class, or a single line comment to explain one little thing. Check out: http://docs.python-guide.org/en/latest/writing/documentation/

It should also not have print statements in it (except the print statement on 272-279 look worth keeping)

So, at a quick glance (not saying I'm right exactly), consider deleting:
- line 323 onwards
- line 319-321
- line 317
- line 315
- line 304 - 306
- line 286 - 297
- line 251 - 254
- line 206 - 240
- line 197 - 201
- ... all Future Feature things
- line 138 - 149 (unused code)
- line 117 - 123 (don't duplicate python code & unused code)
- line 107 - 116 (unused code)
- line 16 - 70

In addition, while going through I noticed:
- get_filename_from_url & save_files seem redundant as you have more code having them as a method than you would if you just wrote the code in the method calling them. This makes readability and understandability hard as I have to jump around to see what's going on.
- if you use with open() you don't need f.close that's the beauty of with open - https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects

Am closing this, once you've pushed up the changes to the feature branch, then you can reopen.

pyMike commented:
Hi Sarah

Thank you for the feedback!

I've been meaning to check out a python guide before continuing.
My goal so far has been to punch out some useful functions in case anyone
needs them
and later package the code into classes, modules and clean things up.

That being said, I'll have a good look at the link you provided before I
make any more changes.

Also, I currently have 3 branches, the last one being the most important
(or the only important one)
All 3 branches are for the same module and at this point the earlier ones
can be disregarded.

My question to you is... should I close the earlier pull requests of the
older branches?

I think it could be a waste of time to look at old code.

Do you know of a good resource that explains this work flow process
comprehensively?
I get the feeling I'm not in sync with the proper work flow process.

Looking forward to getting through this next step.

Hope all is going well

Mike

On Wed, Jul 9, 2014 at 1:56 PM, Sarah Bird notifications@github.com wrote:

> Closed #4 https://github.com/open-contracting/ocdata-blend/pull/4.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/open-contracting/ocdata-blend/pull/4#event-139832997.

pyMike committed:
Remove many print statements and comments. Also removed some reduntant and uneeded statements.
